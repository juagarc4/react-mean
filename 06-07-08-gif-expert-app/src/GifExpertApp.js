import React, { useState } from 'react'
import { AddCategory } from 'components/AddCategory'
import { GifGrid } from 'components/GifGrid'
export default function GifExpertApp({ defaultCategories = [] }) {
  const [categories, setCategories] = useState(defaultCategories)
  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  )
}
