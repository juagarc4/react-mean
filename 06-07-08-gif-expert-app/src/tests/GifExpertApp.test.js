import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import GifExpertApp from 'GifExpertApp'

const wrapper = shallow(<GifExpertApp />)

describe('Tests in <GifExpertApp />', () => {
  test('Should show the component correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Should show a list of categories', () => {
    const categories = ['Dragon Ball', 'Matrix', 'One punch']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})
