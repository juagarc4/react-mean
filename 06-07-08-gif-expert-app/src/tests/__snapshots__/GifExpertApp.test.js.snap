// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> Should show the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
